lib/vrl/stdlib/src/sort.rs

```diff
@@ -0,0 +1,141 @@
+use std::cmp::Ordering;
+
+use vrl::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct Sort;
+
+impl Function for Sort {
+    fn identifier(&self) -> &'static str {
+        "sort"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                kind: kind::ARRAY,
+                required: true,
+            },
+            Parameter {
+                keyword: "reverse",
+                kind: kind::BOOLEAN,
+                required: false,
+            },
+        ]
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[
+            Example {
+                title: "sort array",
+                source: r#"sort([3, 1, 2])"#,
+                result: Ok("[1, 2, 3]"),
+            },
+            Example {
+                title: "reverse sort array",
+                source: r#"sort([3, 1, 2], true)"#,
+                result: Ok("[3, 2, 1]"),
+            },
+        ]
+    }
+
+    fn compile(
+        &self,
+        _state: &state::TypeState,
+        _ctx: &mut FunctionCompileContext,
+        arguments: ArgumentList,
+    ) -> Compiled {
+        let value = arguments.required("value");
+        let reverse = arguments.optional("reverse").unwrap_or(expr!(false));
+
+        Ok(SortFn { value, reverse }.as_expr())
+    }
+}
+
+#[derive(Debug, Clone)]
+struct SortFn {
+    value: Box<dyn Expression>,
+    reverse: Box<dyn Expression>,
+}
+
+impl FunctionExpression for SortFn {
+    fn resolve(&self, ctx: &mut Context) -> Resolved {
+        let mut value = self.value.resolve(ctx)?.try_array()?;
+        let reverse = self.reverse.resolve(ctx)?.try_boolean()?;
+
+        value.sort_by(|a, b| {
```

Comment:
We should probably use `sort_unstable_by` since it is generally faster and doesn't use extra memory. Two equivalent `Value` instances can be swapped without a user being able to tell the difference.